dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/ConvolutionalLayer.java

```diff
@@ -88,30 +100,68 @@ public boolean isLearnable() {
    * @return the converted column.
    */
   private Matrix im2col(final int imageIndex, final Matrix input) {
-    final int[] inputShape = getInputShape();
+    final int inputSize = inputHeight * inputWidth;
+    final int outputSize = outputShape[0] * outputShape[1];
+    final Matrix col =
+        matrixFactory.zeros(kernelHeight * kernelWidth * inputChannel, outputSize);
+    for (int c = 0; c < inputChannel; ++c) {
+      for (int kh = 0; kh < kernelHeight; ++kh) {
+        for (int kw = 0; kw < kernelWidth; ++kw) {
+          int ih = kh - paddingHeight;
+          for (int oh = 0; oh < outputShape[0]; ++oh) {
+            if (ih >= 0 && ih < inputHeight) {
+              int iw = kw - paddingWidth;
+              for (int ow = 0; ow < outputShape[1]; ++ow) {
+                if (iw >= 0 && iw < inputWidth) {
+                  col.put(kh * kernelWidth + kw + c * kernelHeight * kernelWidth, oh * outputShape[1] + ow,
+                      input.get(ih * inputWidth + iw + c * inputSize, imageIndex));
+                }
+                iw += strideWidth;
+              }
+            }
+            ih += strideHeight;
+          }
+        }
+      }
+    }
+    return col;
+  }
+
+  /**
+   * Transform the given image to row form to facilitate matrix multiplication.
```

Comment:
Typo. Transform the given image to row**s**.
Could you also fix typo in the javadoc of `im2col`? Transform the given image to column**s**.